lib/internal/Magento/Framework/Cache/Backend/Redis.php

```diff
@@ -51,7 +51,10 @@ public function load($id, $doNotTestCacheValidity = false)
                 $redis->hGet(self::PREFIX_KEY . $key, self::FIELD_DATA);
             }
 
-            $this->preloadedData = array_filter(array_combine($this->preloadKeys, $redis->exec()));
+            $redisResponse = $redis->exec();
+            $this->preloadedData = $redisResponse ?
```

Comment:
I suggest you check here if the `redisResponse` is an array, then only do the further action otherwise set an empty array.

Reply:
`$redis->exec() `return array so not required.

Reply:
Here the method signature for `exec()` in `Credis_Client` class tells us that it returns `array` only but internally as per the documentation of [credis](https://github.com/colinmollenhour/credis), for better performance wraps the [phpredis](https://github.com/phpredis/phpredis) library when available. but when checking the implementation in phpredis, exec() can return Redis|array|false also. Thats why I asked to add a check here if the return value is array then only move further.